src/wui/fieldaction.cc

```diff
@@ -897,8 +897,8 @@ void FieldActionWindow::building_icon_mouse_in(const Widelands::DescriptionIndex
 						          .highlight_overlapping_workarea_for(d->name(), &positive))) ||
 						    ((descr.type() == Widelands::MapObjectType::MILITARYSITE ||
 						      descr.type() == Widelands::MapObjectType::WAREHOUSE) &&
-						     imm_type != Widelands::MapObjectType::MILITARYSITE &&
-						     imm_type != Widelands::MapObjectType::WAREHOUSE)) {
+						     d->type() != Widelands::MapObjectType::MILITARYSITE &&
```

Comment:
`d = cs->get_info().becomes;` can be `nullptr`, so I suspect that this can crash

Reply:
If a constructionsite does not know what building it is supposed to become, surely there must be something very wrong…

Reply:
Thanks for the assert - I didn't look that deeply into the code